src/workerd/io/actor-cache.c++

```diff
@@ -2155,10 +2155,17 @@ void ActorCache::ensureFlushScheduled(const WriteOptions& options) {
 
   if (!flushScheduled) {
     flushScheduled = true;
-    auto flushPromise = lastFlush.addBranch().attach(kj::defer([this]() {
-      flushScheduled = false;
-      flushScheduledWithOutputGate = false;
-    })).then([this]() {
+    auto flushPromise = lastFlush.addBranch().attach(
+      util::DurationExceededLogger(
```

Comment:
I don't think this works.  `Own<T>` has an attachVal that would work like this, but kj::Promise doesn't have an attachVal.  Maybe you can just heap allocate it and then move it here.